### ISSUE TITLE
Error while running `translations:sync`: ArgumentError: comparison of Pathname with String failed

### ISSUE THREAD

**Issue by 11mdlow** [translation/rails#41]
I came across an issue while running `rails translations:sync` recently:

```
ArgumentError: comparison of Pathname with String failed
/Users/mlow/.rbenv/versions/2.3.7/lib/ruby/gems/2.3.0/gems/translation-1.22/lib/translation_io/client/sync_operation/apply_yaml_source_edits_step.rb:65:in `sort'
/Users/mlow/.rbenv/versions/2.3.7/lib/ruby/gems/2.3.0/gems/translation-1.22/lib/translation_io/client/sync_operation/apply_yaml_source_edits_step.rb:65:in `yaml_sources_reload_needed?'
/Users/mlow/.rbenv/versions/2.3.7/lib/ruby/gems/2.3.0/gems/translation-1.22/lib/translation_io/client/sync_operation/apply_yaml_source_edits_step.rb:47:in `reload_or_reuse_yaml_sources'
/Users/mlow/.rbenv/versions/2.3.7/lib/ruby/gems/2.3.0/gems/translation-1.22/lib/translation_io/client/sync_operation/apply_yaml_source_edits_step.rb:23:in `block in run'
/Users/mlow/.rbenv/versions/2.3.7/lib/ruby/gems/2.3.0/gems/translation-1.22/lib/translation_io/client/sync_operation/apply_yaml_source_edits_step.rb:20:in `each'
/Users/mlow/.rbenv/versions/2.3.7/lib/ruby/gems/2.3.0/gems/translation-1.22/lib/translation_io/client/sync_operation/apply_yaml_source_edits_step.rb:20:in `run'
/Users/mlow/.rbenv/versions/2.3.7/lib/ruby/gems/2.3.0/gems/translation-1.22/lib/translation_io/client/sync_operation.rb:26:in `run'
/Users/mlow/.rbenv/versions/2.3.7/lib/ruby/gems/2.3.0/gems/translation-1.22/lib/translation_io/client.rb:19:in `sync'
/Users/mlow/.rbenv/versions/2.3.7/lib/ruby/gems/2.3.0/gems/translation-1.22/lib/translation_io/tasks.rb:23:in `block (2 levels) in <top (required)>'
/Users/mlow/.rbenv/versions/2.3.7/lib/ruby/gems/2.3.0/gems/rake-12.3.3/lib/rake/task.rb:273:in `block in execute'
/Users/mlow/.rbenv/versions/2.3.7/lib/ruby/gems/2.3.0/gems/rake-12.3.3/lib/rake/task.rb:273:in `each'
/Users/mlow/.rbenv/versions/2.3.7/lib/ruby/gems/2.3.0/gems/rake-12.3.3/lib/rake/task.rb:273:in `execute'
/Users/mlow/.rbenv/versions/2.3.7/lib/ruby/gems/2.3.0/gems/bugsnag-6.7.3/lib/bugsnag/integrations/rake.rb:18:in `execute_with_bugsnag'
/Users/mlow/.rbenv/versions/2.3.7/lib/ruby/gems/2.3.0/gems/rake-12.3.3/lib/rake/task.rb:214:in `block in invoke_with_call_chain'
/Users/mlow/.rbenv/versions/2.3.7/lib/ruby/2.3.0/monitor.rb:214:in `mon_synchronize'
/Users/mlow/.rbenv/versions/2.3.7/lib/ruby/gems/2.3.0/gems/rake-12.3.3/lib/rake/task.rb:194:in `invoke_with_call_chain'
/Users/mlow/.rbenv/versions/2.3.7/lib/ruby/gems/2.3.0/gems/rake-12.3.3/lib/rake/task.rb:183:in `invoke'
/Users/mlow/.rbenv/versions/2.3.7/lib/ruby/gems/2.3.0/gems/rake-12.3.3/lib/rake/application.rb:160:in `invoke_task'
/Users/mlow/.rbenv/versions/2.3.7/lib/ruby/gems/2.3.0/gems/rake-12.3.3/lib/rake/application.rb:116:in `block (2 levels) in top_level'
/Users/mlow/.rbenv/versions/2.3.7/lib/ruby/gems/2.3.0/gems/rake-12.3.3/lib/rake/application.rb:116:in `each'
/Users/mlow/.rbenv/versions/2.3.7/lib/ruby/gems/2.3.0/gems/rake-12.3.3/lib/rake/application.rb:116:in `block in top_level'
/Users/mlow/.rbenv/versions/2.3.7/lib/ruby/gems/2.3.0/gems/rake-12.3.3/lib/rake/application.rb:125:in `run_with_threads'
/Users/mlow/.rbenv/versions/2.3.7/lib/ruby/gems/2.3.0/gems/rake-12.3.3/lib/rake/application.rb:110:in `top_level'
/Users/mlow/.rbenv/versions/2.3.7/lib/ruby/gems/2.3.0/gems/railties-5.2.4.4/lib/rails/commands/rake/rake_command.rb:23:in `block in perform'
/Users/mlow/.rbenv/versions/2.3.7/lib/ruby/gems/2.3.0/gems/rake-12.3.3/lib/rake/application.rb:186:in `standard_exception_handling'
/Users/mlow/.rbenv/versions/2.3.7/lib/ruby/gems/2.3.0/gems/railties-5.2.4.4/lib/rails/commands/rake/rake_command.rb:20:in `perform'
/Users/mlow/.rbenv/versions/2.3.7/lib/ruby/gems/2.3.0/gems/railties-5.2.4.4/lib/rails/command.rb:48:in `invoke'
/Users/mlow/.rbenv/versions/2.3.7/lib/ruby/gems/2.3.0/gems/railties-5.2.4.4/lib/rails/commands.rb:18:in `<top (required)>'
bin/rails:11:in `require'
bin/rails:11:in `<main>'
Tasks: TOP => translation:sync
```

I did a little digging and it seems like `Pathname` objects are being added to the `I18n.load_path` somehow. In this case there were a couple:
* a yml file internal to our project, which actually had another entry of type `String` with the same path
* a yml entry from an external gem: https://github.com/dry-rb/dry-validation/tree/v0.12.3. The file in question is `config/errors.yml` from that particular tag/version

These files get added to the load path because `File.exist?` and `File.extname` can take in a `String` or a `Pathname`, so I'm not entirely sure what the best route is here. Let me know what you think!

**Comment by MichaelHoste** [comment]
Hi @11mdlow,

Thank you for finding this issue and for the pull request!

My guess is that Pathnames should not be part of `I18n.load_path`. It seems to be an issue caused by https://github.com/dry-rb/dry-validation because I found another application having a similar problem: https://github.com/AlchemyCMS/alchemy_cms/issues/1464

Maybe they fixed it in newer releases?

However, since it's a small fix that can avoid similar issues, I fixed it here: https://github.com/translation/rails/commit/ed701490f875c7e250d0ef2370826f974547908c

It's the same approach as your pull request but I applied `to_s` higher in the stack. I also applied a `uniq` since you said that you noticed 2 similar paths in your project. Better be safe 😄 

Could you please try using the `master` branch in your project and confirm that it works?

If so, I'll release a new version.

**Comment by 11mdlow** [comment]
Hi @MichaelHoste, thanks for the quick response. That seems to have fixed it! 

FWIW I suspect rails/i18n is doing something weird during app initialization when it sees a `.yml` file, because in both instances there is a direct reference to the file in the code. In dry-validation it's here: https://github.com/dry-rb/dry-validation/blob/e740dd270aa386163eda394d62546464903b690c/lib/dry/validation/constants.rb#L19. I think `I18n` allows `Pathname` objects, which is probably why it's never been addressed in that gem.

Anyways, thanks for the quick fix!

**Comment by MichaelHoste** [comment]
Thank you for your feedback!

Since it seems to work, I released [version 1.23](https://github.com/translation/rails/releases/tag/v1.23) of the gem.

It's not clear from [the I18n API](https://www.rubydoc.info/github/svenfuchs/i18n/master/I18n%2FConfig:load_path) if `I18n.load_path` accepts anything other than a String.
In doubt, using `to_s` is certainly a best practice when using it 😄
